### PR TITLE
Update code of conduct

### PR DESCRIPTION
**Pull request by Rozzii**
This commit:
  - Aligns Metal3-io code of conduct with CNCF code of conduct v1.3 as recommended in [CNCF Annual Report Review](https://docs.google.com/document/d/14KObDX-ybKzgZ6sjpfFXNyNOmeWIoIdfd7tCtzwVNhk/edit#heading=h.3ff1v7xs75xa)

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,78 +1,85 @@
...
-## Scope
+## Reporting
 
-This Code of Conduct applies both within project spaces and in public spaces
-when an individual is representing the project or its community. Examples of
-representing a project or community include using an official project e-mail
-address, posting via an official social media account, or acting as an appointed
-representative at an online or offline event. Representation of a project may be
-further defined and clarified by project maintainers.
+For incidents occurring in the Kubernetes community, contact the [Kubernetes Code of Conduct Committee](https://git.k8s.io/community/committee-code-of-conduct) via <conduct@kubernetes.io>. You can expect a response within three business days.
```
Comment (tuminoid) [inline CODE_OF_CONDUCT.md:64]: We want to update our contacts here.

**Review by tuminoid — CHANGES_REQUESTED** [review]
We need to adapt this to Metal3, like the contacts etc.

Where was this sourced from?

**Comment by tuminoid** [comment]
As it is recommended by the CNCF to align CoC, I'm for it. 

> We need to adapt this to Metal3, like the contacts etc.
> 
> Where was this sourced from?

Nevermind, I re-read the thing and its OK. The source is CNCF as-is, and not k8s adapted one as I had though.

**Review by tuminoid — APPROVED** [review]
/approve
but also
/hold
Let's discuss this in community before merging.

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -1,78 +1,85 @@
...
-## Scope
+## Reporting
 
-This Code of Conduct applies both within project spaces and in public spaces
-when an individual is representing the project or its community. Examples of
-representing a project or community include using an official project e-mail
-address, posting via an official social media account, or acting as an appointed
-representative at an online or offline event. Representation of a project may be
-further defined and clarified by project maintainers.
+For incidents occurring in the Metal3 community, contact the `Metal3 maintaner e-mail list` via <metal3-maintainers@googlegroups.com>. You can expect a response within three business days.
```
Comment (tuminoid) [inline CODE_OF_CONDUCT.md:64]: As said, after re-reading the wording again, I don't think we're supposed to handle the CoC reports ourselves, but have CNCF handle them. This is clear distinction between the original CoC and CNCF CoC. Kubernetes here is just mentioned as an exception as it is such large project that they have separate CoC handling.

I also checked few other CNCF projects too. Not all of them implement CNCF CoC, but those who do did not change the CoC like this. This is of course up to discussion, but IMO we should keep the document as-is and not take the responsibility to ourselves to handle CoC, but leave it to CNCF.

Thoughts @kashifest ?

**Review by smoshiur1237 — APPROVED** [review]
Thanks for the doc and it looks much better now.
/lgtm

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (kashifest) [reply CODE_OF_CONDUCT.md]: I agree with @tuminoid's reasoning

**Comment by Rozzii** [comment]
As per off GitHub discussion we will just provide a link to CNCF CoC instead of copying it.

**Pull request merged**